src/hooks/useProduct.ts

```diff
@@ -20,12 +20,17 @@ const useProductInfo: UseProductInfo = ({ item }) => {
 
   const itemName = productSwatch?.selectedVariation?.itemName || item.itemName;
   const itemPrice = productSwatch?.selectedVariation?.price || getPrice(item);
-  const salePrice = productSwatch?.selectedVariation?.salePrice || getSalePrice(item);
   const itemImageUrl = productSwatch?.selectedVariation?.imageUrl || item.imageUrl;
   const itemUrl = productSwatch?.selectedVariation?.url || item.url;
   const variationId = productSwatch?.selectedVariation?.variationId;
   const { itemId } = item;
 
+  function isValidSalePrice(salePrice: number, usualPrice: number) {
+    return salePrice && usualPrice && salePrice > 0 && salePrice < usualPrice;
+  }
+  let salePrice = productSwatch?.selectedVariation?.salePrice || getSalePrice(item);
```

Comment:
small lint thing: newline after function declaration
```suggestion
  }
  
  let salePrice = productSwatch?.selectedVariation?.salePrice || getSalePrice(item);
```